gitgud/test_operator.py

```diff
@@ -0,0 +1,54 @@
+import pytest
+
+from gitgud.skills import all_skills
+from gitgud.operations import get_operator
+
+
+@pytest.fixture
+def file_operator(gg):
+    file_operator = get_operator()
+    assert file_operator is not None
+    return file_operator
+
+
+@pytest.fixture(scope='function')
+def progress_data(file_operator, gg):
+    progress_data = file_operator.read_progress_file()
+    return progress_data
+
+
+def test_get_level_progress(file_operator, progress_data):
+    level = all_skills["1"]["1"]
+    assert file_operator.get_level_progress(level) in {
+        "unvisited", "visited", "partial", "complete"
+    }
+
+
+def test_update_progress_file(file_operator, progress_data):
+    file_operator.read_progress_file()
+    level = all_skills["1"]["1"]
+    progress_data[level.skill.name].update({level.name: "value"})
+    file_operator.update_progress_file(progress_data)
+    assert file_operator.get_level_progress(level) == "value"
```

Comment:
Is this safe to do? Is there a possibility the dictionary will show "value" during other tests?

Reply:
Yep, this is 100% safe to do. `value` won't appear in other tests because the fixture is destroyed at the end of the test function and is rebuilt for the other tests, since it has function scope. (Actually, `pytest.fixture(scope='function')` is redundant, since [it's the default](https://docs.pytest.org/en/stable/fixture.html#fixture-scopes) -- I will fix that decorator). Here's a short [demonstration](https://gist.github.com/sahansk2/07d374a4e96432775be69f10efb70612) of what happens.